src/core/createCookieProxy.js

```diff
@@ -12,6 +12,9 @@ governing permissions and limitations under the License.
 
 import cookie from "../utils/cookie";
 import getTopLevelCookieDomain from "../utils/getTopLevelCookieDomain";
+import memoize from "../utils/memoize";
+
+const memoizedGetTopLevelCookieDomain = memoize(getTopLevelCookieDomain);
```

Comment:
We could create `memoizedGetTopLevelCookieDomain` in a few different places:

1. Right here like we are now. This is really only useful if we only end up using it in this module.
2. We could create a utility called `memoizedGetTopLevelCookieDomain.js` that would live beside `getTopLevelCookieDomain.js`.
3. We could create `memoizedGetTopLevelCookieDomain` up in `main.js` and pass it into `createCookieProxy.js`, and later pass it down to all components if they end up needing it.

As I think about this again, I think I'm going to add a commit that changes it to number 3.